backend/src/payments/payments.controller.ts

```diff
@@ -52,25 +65,50 @@ export class PaymentsController {
   @Get('trainees/:traineeId')
   async findAllByTraineeId(
     @Param('traineeId', ParseIntPipe) traineeId: number,
+    @Query() query: PaymentFindAllQueryDto,
   ): Promise<PaymentFindAllByTraineeResponseDto> {
-    const payments = await this.paymentsService.findAllByTraineeId(traineeId);
-
-    return { payments };
+    const { payments, count } = await this.paymentsService.findAllByTraineeId(
+      traineeId,
+      {
+        pagination: {
+          page: query.page,
+          pageSize: query.pageSize,
+        },
+        sort: {
+          sortBy: query.sortBy,
+          sortOrder: query.sortOrder,
+        },
+        filter: query.filters ?? {},
+      },
+    );
+    return { payments, total: count };
   }
 
-  @Roles(UserRole.Trainee)
+  @Roles(UserRole.Trainee, UserRole.Admin)
```

Comment:
Admin shouldn't be added here as they will never have any payments assigned to them